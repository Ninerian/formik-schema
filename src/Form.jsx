import React from 'react';

const widget = (config) => (
  <input type={config.type} className="form-control" id={config.name} placeholder={config.placeholder} />
);

const buildFieldRow = (config) => (
  <div key={config.name} className="form-group row">
    <label for={config.name} className="col-sm-2 col-form-label">{config.title}</label>
    <div className="col-sm-10">
      {widget(config)}
    </div>
  </div>
);

export const buildForm = (schema) => (
  <form>
    {schema.fields.map(buildFieldRow)}
  </form>
);
