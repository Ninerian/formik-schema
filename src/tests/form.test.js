import React from 'react';
import { buildForm } from '../Form.jsx';
import renderer from 'react-test-renderer';

// See https://getbootstrap.com/docs/4.0/components/forms/#horizontal-form
test('Generates an horizontal form', () => {
  const form = buildForm({
    fields: [
      {
        name: "email",
        title: "Email",
        type: "email",
      },
      {
        name: "password",
        title: "Mot de passe",
        type: "password",
      },
      {
        name: "accept",
        title: "J'accepte.",
        type: "opt-in",
      },
    ],
  });
  const component = renderer.create(form);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
