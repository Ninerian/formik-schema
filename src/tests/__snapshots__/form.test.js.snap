// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generates an horizontal form 1`] = `
<form>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-2 col-form-label"
      for="email"
    >
      Email
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        id="email"
        placeholder={undefined}
        type="email"
      />
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-2 col-form-label"
      for="password"
    >
      Mot de passe
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        id="password"
        placeholder={undefined}
        type="password"
      />
    </div>
  </div>
  <div
    className="form-group row"
  >
    <label
      className="col-sm-2 col-form-label"
      for="accept"
    >
      J'accepte.
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        id="accept"
        placeholder={undefined}
        type="opt-in"
      />
    </div>
  </div>
</form>
`;
